tests/analysis/trials/test_has_passed_step_scenarios.py

```diff
@@ -38,6 +76,110 @@ def test_single_writer_all_steps_written_complete_job():
     assert trial.last_complete_step == 6
 
 
+@pytest.mark.slow
+def test_single_writer_all_steps_written_complete_job_two_modes():
+    """Test Scenario Description"
+     workers : [a]
+     modes: TRAIN, EVAL
+     steps :{
+        0: [worker:a, mode: TRAIN, mode_step: 0],
+        10: [worker:a, mode: TRAIN, mode_step: 10],
+        20: [worker:a, mode: TRAIN, mode_step: 20],
+        30: [worker:a, mode: TRAIN, mode_step: 30],
+        40: [worker:a, mode: EVAL, mode_step: 0],
+        50: [worker:a, mode: EVAL, mode_step: 10],
+        60: [worker:a, mode: EVAL, mode_step: 20],
+        70: [worker:a, mode: EVAL, mode_step: 30]
+        }
+    END_OF_JOB.ts --> Present
+    """
+
+    path = os.path.join("ts_output/train/", str(uuid.uuid4()))
+    dummy_trial_creator(trial_dir=path, num_workers=1, job_ended=True)
+    for i in range(0, 31, 10):
+        dummy_step_creator(
+            trial_dir=path, global_step=i, mode="TRAIN", mode_step=i, worker_name="worker_0"
+        )
+
+    for i in range(0, 31, 10):
+        dummy_step_creator(
+            trial_dir=path, global_step=i + 40, mode="EVAL", mode_step=i, worker_name="worker_0"
+        )
+
+    trial = create_trial(path)
+    num_workers = len(trial.workers())
+    assert num_workers == 1
+    assert trial.loaded_all_steps is True
+    all_steps = trial.steps(show_incomplete_steps=True)
+    completed_steps = trial.steps()
+    assert all_steps == [0, 10, 20, 30, 40, 50, 60, 70]
```

Comment:
move this to end of test or another test, There should be unit test only to check has_passed_step , 
create data, create trail, assert has_passed_step